### PR TITLE
DEV: Convert min_trust_to_edit_wiki_post to groups

### PR DESCRIPTION
**Pull request by lis2**
This change converts the min_trust_to_edit_wiki_post site setting to edit_wiki_post_allowed_groups.

See: https://meta.discourse.org/t/283408

- Hides the old setting
- Adds the new site setting
- Adds a deprecation warning
- Updates to use the new setting
- Adds a migration to fill in the new setting if the old setting was changed 
- Adds an entry to the site_setting.keywords section 
- Updates tests to account for the new change

After a couple of months, we will remove the email_in_min_trust setting entirely.

Internal ref: /t/117248

### CODE REVIEW

File: config/locales/server.en.yml [hunk 1]
```diff
@@ -1961,6 +1961,7 @@ en:
     allow_flagging_staff: "If enabled, users can flag posts from staff accounts."
 
     min_trust_to_edit_wiki_post: "The minimum trust level required to edit post marked as wiki."
+    edit_wiki_post_allowed_groups: "Groups that are allowed to edit post marked as wiki."
```
Comment (martin-brennan) [inline config/locales/server.en.yml:1964]:
```suggestion
    edit_wiki_post_allowed_groups: "Groups that are allowed to edit posts marked as wiki."
```

The old one was wrong too, but post should be posts.

**Review by martin-brennan — APPROVED** [review]
LGTM apart from minor copyedit :)

File: config/locales/server.en.yml [hunk 2]
```diff
@@ -2550,6 +2551,7 @@ en:
       approve_unless_allowed_groups: "approve_unless_trust_level"
       approve_new_topics_unless_allowed_groups: "approve_new_topics_unless_trust_level"
       email_in_allowed_groups: "email_in_min_trust"
+      edit_wiki_post_allowed_groups: "minmin_trust_to_edit_wiki_post"
```
Comment (Drenmi) [inline config/locales/server.en.yml:2554]: There's a copy-paste error here:

```
"minmin_trust_to_edit_wiki_post"
```

should be:

```
"min_trust_to_edit_wiki_post"
```

**Review by Drenmi — COMMENTED** [review]
I think we may have missed a hard-coded line here:

https://github.com/discourse/discourse/blob/main/lib/guardian/post_guardian.rb#L333

I suspect this is meant to be using the same setting, but was never changed to use `min_trust_to_edit_wiki_post` in the first place. 🙂

**Pull request merged**